test/cli/detailed-errors-completeness/test.out

```diff
@@ -0,0 +1,261 @@
+      `Integer` is not a subtype of `String` for item `0`
+
+test.rb:68: Return type `[String]` does not match return type of overridable method `A#foo` https://srb.help/5035
+    68 |  def foo(x, y:, z: [''], &blk)
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    test.rb:50: Super method defined here with return type `[Integer]`
+    50 |  def foo(x, y:, z: [1], &blk)
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+  Note:
+    A method's return type must be a subtype of the return type on the super method.
+  Possible explanation:
+    `String` is not a subtype of `Integer` for item `0`
+
+test.rb:73: Parameter **`rest` of type `[String]` not compatible with type of overridable method `A#bar` https://srb.help/5035
+    73 |  def bar(**rest)
+          ^^^^^^^^^^^^^^^
+    test.rb:56: The super method parameter **`rest` was declared here with type `[Integer]`
+    56 |  def bar(**rest)
+          ^^^^^^^^^^^^^^^
+  Note:
+    A parameter's type must be a supertype of the same parameter's type on the super method.
+  Possible explanation:
+    `Integer` is not a subtype of `String` for item `0`
+
+test.rb:7: Expected `[Integer]` but found `[String("")]` for method result type https://srb.help/7005
+     7 |  ['']
+          ^^^^
+  Expected `[Integer]` for result type of method `foo`:
+    test.rb:6:
+     6 |def foo(x)
+        ^^^^^^^^^^
+  Got `[String("")] (1-tuple)` originating from:
+    test.rb:7:
+     7 |  ['']
+          ^^^^
+  Possible explanation:
+    `String("")` is not a subtype of `Integer` for item `0`
+
+test.rb:10: Expected `[Integer]` but found `[String("")]` for argument `x` https://srb.help/7002
+    10 |foo([''])
+            ^^^^
+  Expected `[Integer]` for argument `x` of method `Object#foo`:
+    test.rb:5:
+     5 |sig { params(x: [Integer]).returns([Integer]) }
+                     ^
+  Got `[String("")] (1-tuple)` originating from:
+    test.rb:10:
+    10 |foo([''])
+            ^^^^
+  Possible explanation:
+    `String("")` is not a subtype of `Integer` for item `0`
+
+test.rb:17: Expected `[Integer]` for keyword parameter `x` but found `[String]` from keyword splat https://srb.help/7002
+    17 |bar(**h)
+            ^^^
+  Expected `[Integer]` for argument `x` of method `Object#bar`:
+    test.rb:12:
+    12 |sig { params(x: [Integer]).void }
+                     ^
+  Got `T::Hash[Symbol, [String]]` originating from:
+    test.rb:16:
+    16 |h = T.let({}, T::Hash[Symbol, [String]])
+            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    test.rb:17:
+    17 |bar(**h)
+            ^^^
+  Note:
+    A `Hash` passed as a keyword splat must match the type of all keyword parameters
+    because Sorbet cannot see what specific keys exist in the `Hash`.
+  Possible explanation:
+    `String` is not a subtype of `Integer` for item `0`
+  Autocorrect: Use `-a` to autocorrect
+    test.rb:17: Replace with `T.unsafe(h)`
+    17 |bar(**h)
+              ^
```

Comment:
This error message is kind of cursed actually. The problem is actually that Sorbet doesn't even know if `h` has an `x` key at all.

It seems like we want to get rid of the error note here and move that into the error details collector?